### PR TITLE
Enable new reserve to work with higher max rate ( PRECISION * 10^7). in contracts KyberReserveHighRate + enhanced steps contract.

### PR DESCRIPTION
**Pull request by ilanDoron**
Current ConversionRateEnhancedSteps inherits MAX_RATE from utils contract and uses it in addBps function.

currently our code uses PRECISION * 10^6
but in Katalyst MAX_RATE is PRECISION * 10^7 
for using the bigger value. this PR updates MAX_RATE in use for two contracts:
KyberReseve - creates a new contract KyberReserveHighRate
and updating enhanced steps contract, which can be updated since not yet used in production.

code was written to enable max code reuse and minimal code duplication

Wrapper for enhanced conversion rates was modified:
1. remove call to setQtyStepFunction which can't be used in enhanced conversion rate.
2. call set imbalance with x and y arrays with different length.

### CODE REVIEW

File: contracts/mock/ConversionRatesEnhancedOpen.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import "../reserves/fprConversionRate/ConversionRateEnhancedSteps.sol";
+
+
+contract ConversionRateEnhancedOpen is ConversionRateEnhancedSteps {
+
+    function ConversionRateEnhancedOpen(address _admin) public 
+        ConversionRateEnhancedSteps(_admin)
+        { } // solhint-disable-line no-empty-blocks
+
+    function getRateOpenData(
```
Comment (manhlx3006) [inline contracts/mock/ConversionRatesEnhancedOpen.sol:12]: Can you give some details about this function?

File: contracts/mock/ConversionRatesEnhancedOpen.sol [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        ERC20 token, 
+        uint currentBlockNumber, 
+        bool buy, 
+        uint qty 
+    ) public view returns(
+        uint rateWithSteps,
+        uint rateWithoutSteps,
+        int extraBpsYdata
+    ) {
+        // get rate update block
```
Comment (manhlx3006) [inline contracts/mock/ConversionRatesEnhancedOpen.sol:22]: Should we check if token is listed?

File: contracts/mock/ConversionRatesEnhancedOpen.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        if (currentBlockNumber >= updateRateBlock + validRateDurationInBlocks) return (0, 0, 0); // rate is expired
+        // check imbalance
+        int totalImbalance;
+        int blockImbalance;
+        (totalImbalance, blockImbalance) = getImbalance(token, updateRateBlock, currentBlockNumber);
+
+        return getRateDataFakeImbalance(token, buy, qty, totalImbalance);
+    }
+
+    function getRateDataFakeImbalance(
```
Comment (manhlx3006) [inline contracts/mock/ConversionRatesEnhancedOpen.sol:35]: We don't have check for `totalImbalance` (or `fakeImbalance`) and `blockImbalance`, is it our intention?

File: contracts/mock/ConversionRatesEnhancedOpen.sol [hunk 4]
```diff
@@ -2,13 +2,24 @@ pragma solidity 0.4.18;
...
+ *      - get rate queries that also show steps data (accumulated Y value).
+ *      - query get rate with fake imbalance.
+ */
 contract ConversionRateEnhancedOpen is ConversionRateEnhancedSteps {
 
     function ConversionRateEnhancedOpen(address _admin) public 
         ConversionRateEnhancedSteps(_admin)
         { } // solhint-disable-line no-empty-blocks
 
+    ///@dev enables calling get rate and watching extra BPS from step functio.
```
Comment (manhlx3006) [inline contracts/mock/ConversionRatesEnhancedOpen.sol:19]: functio -> function

File: contracts/reserves/fprConversionRate/ConversionRateEnhancedSteps.sol [hunk 5]
```diff
@@ -12,6 +12,7 @@ import "./ConversionRates.sol";
 
 contract ConversionRateEnhancedSteps is ConversionRates {
 
+    uint  constant internal MAX_RATE  = (PRECISION * 10**7); // up to 10M tokens per ETH
```
Comment (Anyhowclick) [inline contracts/reserves/fprConversionRate/ConversionRateEnhancedSteps.sol:15]: extra spacing here, can remove

**Comment by ilanDoron** [comment]
will be merged in another PR

**Pull request closed without merging**